tests/integration/asyncio/reconnect_test.py

```diff
@@ -0,0 +1,269 @@
+import asyncio
+import sys
+import unittest
+
+from hazelcast.asyncio import HazelcastClient
+from hazelcast.errors import HazelcastError, TargetDisconnectedError
+from hazelcast.lifecycle import LifecycleState
+from hazelcast.util import AtomicInteger
+from tests.integration.asyncio.base import HazelcastTestCase
+from tests.util import event_collector
+
+
+class ReconnectTest(unittest.IsolatedAsyncioTestCase, HazelcastTestCase):
+    rc = None
+
+    def setUp(self):
+        self.rc = self.create_rc()
+        self.cluster = self.create_cluster(self.rc)
+
+    async def asyncTearDown(self):
+        await self.shutdown_all_clients()
+        self.rc.exit()
+
+    async def test_start_client_with_no_member(self):
+        with self.assertRaises(HazelcastError):
+            await self.create_client(
+                {
+                    "cluster_members": [
+                        "127.0.0.1:5701",
+                        "127.0.0.1:5702",
+                        "127.0.0.1:5703",
+                    ],
+                    "cluster_connect_timeout": 2,
+                }
+            )
+
+    async def test_start_client_before_member(self):
+        async def run():
+            await asyncio.sleep(1.0)
+            await asyncio.to_thread(self.cluster.start_member)
```

Comment:
Can you port `self.cluster.start_member` to be async so you do not need to spawn this thread?

Reply:
It's out of scope for this work.

`asyncio.to_thread` was introduced to deal with blocking code.
Why do you want to run `cluster_member` as an asyncio task?

Reply:
i like to replace the line with `await self.cluster.start_member()` so not use thread at all.